### ISSUE TITLE
[REVIEW]: rgugik: Search and Retrieve Spatial Data from the Polish Head Office of Geodesy and Cartography in R

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2948]
**Submitting author:** @kadyb (<a href="http://orcid.org/0000-0002-8614-3816">Krzysztof Dyba</a>)
**Repository:** <a href="https://github.com/kadyb/rgugik" target ="_blank">https://github.com/kadyb/rgugik</a>
**Version:** v0.2.1
**Editor:** @kbarnhart
**Reviewer:** @adamhsparks, @mikerspencer
**Archive:** <a href="https://doi.org/10.5281/zenodo.4606706" target="_blank">10.5281/zenodo.4606706</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/72de1e21e7ee7e45e7d7cd8d838158ee/status.svg)](https://joss.theoj.org/papers/72de1e21e7ee7e45e7d7cd8d838158ee)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/72de1e21e7ee7e45e7d7cd8d838158ee"><img src="https://joss.theoj.org/papers/72de1e21e7ee7e45e7d7cd8d838158ee/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/72de1e21e7ee7e45e7d7cd8d838158ee/status.svg)](https://joss.theoj.org/papers/72de1e21e7ee7e45e7d7cd8d838158ee)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/kadyb/rgugik" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@adamhsparks & @mikerspencer, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kbarnhart know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kadyb/rgugik">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kadyb) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/kadyb/rgugik">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@kadyb) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @adamhsparks, @mikerspencer it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.32614/RJ-2018-009 is OK
- 10.2830/63132 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @mikerspencer, please update us on how your review is going.

**Comment by whedon** [comment]
:wave: @adamhsparks, please update us on how your review is going.

**Comment by kbarnhart** [comment]
@mikerspencer and @adamhsparks, we have an automatic reminder two weeks into the review. Feel free to provide an update if you have one. And if you have any issues as you work on your reviews, please let me know. Sometimes reviewers have permissions issues with being able to check the checkboxes, if that comes up for you, just let me know and I should be able to fix it.

**Comment by adamhsparks** [comment]
I have to admit, I'm a bit slow off the mark here and can't remember what I've actually done. If I check this invitation, https://github.com/openjournals/joss-reviews/invitations GitHub says it's expired. Did I manage to accept before that happened?

**Comment by kbarnhart** [comment]
@whedon re-invite @adamhsparks  as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@adamhsparks  please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by kbarnhart** [comment]
This should be a fresh invite link.  Let me know if you have issues with check-boxes after using it.

**Comment by adamhsparks** [comment]
Ok, that's great. Now the tick boxes work for me. 🙏

**Comment by kbarnhart** [comment]
@mikerspencer and @adamhsparks, a friendly reminder that we are nearing the end of the six week period in which JOSS requests you complete your reviews. If you have any questions, please let me know.

**Comment by adamhsparks** [comment]
Thanks for the reminder. This is on my to-do list for today.

**Comment by adamhsparks** [comment]
Overall the package is well written and I'm impressed with the real-world use-case examples in the vignettes. This will be a great benefit to the community by providing an easy to use resource for a reproducible workflow that centralises data fetching for this data resource. Other tools that offer access to this resource are discussed, along with the advantages of this tool over them.

The contribution guidelines are some of the most clear I've seen. Well done!

I've opened a few issues in the repository for the code or documentation issues I've found or other thoughts.

Here are some more specific comments on the paper and package.
1. I think that the paper would benefit from some edits for clarity and flow, so I've not ticked that off yet. I found the writing to be mostly clear, all of the content was covered as required, but in some places the language was awkward and hard to follow.
2. Two functions have zero test coverage, `geonames_download()` and `borders_download()`, the reason for this isn't clear to me.
3. I noted that the test coverage in the README badge, ~87% and paper, don't match what I get locally, ~80%. I'm unsure of the reasons for such a discrepancy?
4. There are four vignettes in the pkgdown website, but only three in the package as installed from the master branch?
5. In several places "digital elevation model" is spelled as "Digital Elevation Model", this is not correct, it is not a proper noun.
6. The package passes all the normal R CMD checks and whatnot and has even used CI to run lintr on the code, which really shows effort put into writing good code. 🙏

**Comment by kbarnhart** [comment]
@adamhsparks thanks for your review. @kadyb if discussion is necessary regarding any of the comments I'd recommend making an issue in the main repository. Otherwise you can respond here. 

@mikerspencer a friendly reminder to provide an update on your review. 

Thanks again to both of you for contributing to the JOSS review process.

**Comment by mikerspencer** [comment]
Thanks @kbarnhart. It's amazing 6 weeks have passed so quickly. My son's school started back today, so I've suddenly got a normal amount of time. I'll work through the review and get it back today or tomorrow.

**Comment by kbarnhart** [comment]
@mikerspencer thanks for letting me know. At JOSS, we aim to be very understanding that the pandemic has changed people's time availability (e.g. , [see this blog post from early in the pandemic](https://blog.joss.theoj.org/2020/05/reopening-joss)). Just keep me posted if you need more time beyond this week.

**Comment by mikerspencer** [comment]
@kbarnhart I've finished my review. There's a lot to commend, especially how clearly the package is structured and documented.

I've raised a couple of minor issues (#61, #62 & #63), but unfortunately I was unable to test functionality due to issue #64. Issue #64 is a connection error when downloading datasets, which consistently occurs at ~15 MB through a download. The issue doesn't affect smaller downloads, e.g. the 2.4 MB files in the automated tests. I've tested at two different times of day and am reasonably sure my internet connection is stable. I'm unsure what to recommend, but have suggested adding larger files to the automated download tests.

**Comment by kbarnhart** [comment]
@mikerspencer thanks for doing this review. 

@kadyb, when you think you have addressed all of the points the reviewers have raised, please ping them and me here so they can re-visit their review checklists. 

If you have any questions for me, please tag me on this issue (or in the in-repo issues).

**Comment by kadyb** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: :point_left:

**Comment by kadyb** [comment]
Hello, @mikerspencer thanks for the review and praise! I would like to respond here to the downloading problem you described:

> (...) unfortunately I was unable to test functionality due to issue 64. Issue 64 is a connection error when downloading datasets, which consistently occurs at ~15 MB through a download. The issue doesn't affect smaller downloads, e.g. the 2.4 MB files in the automated tests. I've tested at two different times of day and am reasonably sure my internet connection is stable.

Both reviewers confirmed this problem. On our side it is hard to reproduce, but the error is certainly there. We think it rather affects foreign users (we suppose it could be due to lower server priority). Nevertheless, thanks to the very active and helpful attitude of the reviewers, we tested two solutions to prevent this problem:
1. All download functions in the package use the `utils::download.file()` backend, which supports different download methods. The default method failed, however reviewers confirmed that changing the method helps to resolve the issue. We have therefore corrected the error returned to clearly inform the user that they can change the method (https://github.com/kadyb/rgugik/pull/68).
2. In the long run (after review), we will add another independent download tool (`curl::curl_download()`) as a backend. However, we detected a problem on Windows (it has been [reported](https://github.com/jeroen/curl/issues/238) to the package developer) and it requires more testing to ensure smooth operation (user can test this using the development version of the package).

More information can be found [here](https://github.com/kadyb/rgugik/issues/64). We believe that the actions taken are sufficient.

**Comment by mikerspencer** [comment]
@kadyb good work on the solution. Which version should I test to complete my review? I presume the updated error returned in not yet on CRAN?

**Comment by kadyb** [comment]
@mikerspencer, yes, you're right. Currently, all changes have been made to the development version on GitHub.

Please use the code below to install the development version (this will allow to test the first proposed solution):

```r
# install.packages("remotes")
remotes::install_github("kadyb/rgugik")
```

To test the second solution (downloading files via `curl` package) , please use this: 

```r
remotes::install_github("kadyb/rgugik@usecurl")
```

**Comment by mikerspencer** [comment]
@kbarnhart I've finished my review. It's a really solid piece of work and was a pleasure to review. Well done to @kadyb.

**Comment by danielskatz** [comment]
@mikerspencer - I didn't realize you were reviewing this submission as well - now I feel bad for bugging you on the other one, but not bad enough that I'm not semi-subtly reminding you about that one too :)

**Comment by mikerspencer** [comment]
@danielskatz Yes, a little bit of awkward timing that the other one came back at a similar time. Don't worry, I've not forgotten - it's in my home tabs, so I see it each time I open a browser session. I'd like to get it off my desk this week, for my own sanity along with yours and the authors!

**Comment by kadyb** [comment]
@mikerspencer, thank you very much for your time, all comments and kind words. Good luck with the second review!

**Comment by adamhsparks** [comment]
Code all looks good. I've added comments in the PDF that I've attached here for my comments on the paper as requested.

[10.21105.joss.02948.pdf](https://github.com/openjournals/joss-reviews/files/6136046/10.21105.joss.02948.pdf)

**Comment by kadyb** [comment]
@adamhsparks, thanks. I corrected all these shortcomings in https://github.com/kadyb/rgugik/commit/265692224a7b6398cdec68f7c89cc0c39347ad7c. Test coverage should be ~87% (https://github.com/kadyb/rgugik/commit/ee186247c3fd885bd61034a7fe53a3171b12b3b0).

**Comment by kbarnhart** [comment]
@adamhsparks and @mikerspencer thanks so much for revisiting your reviews after @kadyb made revisions. 

@adamhsparks it looks like there are a couple of remaining check boxes on your list. If these issues have been addressed, please check them off. Otherwise, let @kadyb know what is needed to address them.

**Comment by adamhsparks** [comment]
@kbarnhart, since @kadyb has just addressed my comments I can now go back and check everything and will update.

**Comment by adamhsparks** [comment]
The paper's grammar looks good now but the test coverage, both with the badge and with the paper is still an issue. I just pulled the latest and ran `covr::report()` and got this result of ~80% coverage. Not 87% or 89%?
<img width="1095" alt="Screen Shot 2021-03-15 at 10 06 59 am" src="https://user-images.githubusercontent.com/3195906/111094712-d02a4b80-8576-11eb-9c7b-20adc0097653.png">

**Comment by kadyb** [comment]
`geonames_download()` and `borders_download()` functions are just for data download and do not contain any complex logic. They are not automatically tested because the downloaded files are large (e.g., 375 MB) and server space is limited. We use [this](https://github.com/kadyb/rgugik/blob/master/.github/workflows/test-coverage.yml) configuration file to calculate the test coverage value on [codecov.io](https://codecov.io/gh/kadyb/rgugik). Please run this code to get identical results locally:

```r
fun_excl = c("geonames_download", "borders_download")
cov = covr::package_coverage(function_exclusions = fun_excl)
cov
#> rgugik Coverage: 87.55%
```

**Comment by adamhsparks** [comment]
Ah, thanks for the explanation.

I’m curious, why not use the `#nocov` tags in these files to exclude coverage if they aren’t/don’t need to be tested?

**Comment by adamhsparks** [comment]
@kbarnhart, @kadyb has done a fantastic job with this. I've ticked off the remaining boxes (my previous question notwithstanding).

**Comment by kadyb** [comment]
> I’m curious, why not use the `#nocov` tags in these files to exclude coverage if they aren’t/don’t need to be tested?

I could be wrong, but I think the preferred way is to use the `function_exclusions` argument. However, there are many ways to achieve the effect.

@adamhsparks, thanks for your time and helpful comments!

**Comment by kadyb** [comment]
@whedon generate pdf

**Comment by adamhsparks** [comment]
Thanks, in that case I may be doing it wrong! I’ve learned something new doing this review.

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: :point_left:

**Comment by kbarnhart** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.32614/RJ-2018-009 is OK
- 10.2830/63132 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kbarnhart** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: :point_left:

**Comment by kbarnhart** [comment]
@kadyb at this point could you:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by kadyb** [comment]
1. https://github.com/kadyb/rgugik/releases/tag/v0.2.1 (version tag: `v0.2.1`)
2. https://zenodo.org/record/4606706
3. Done.
4. https://doi.org/10.5281/zenodo.4606705 (`10.5281/zenodo.4606706`)

If something is wrong, please let me know.

**Comment by kbarnhart** [comment]
@whedon set version as v0.2.1

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by kbarnhart** [comment]
@whedon set v0.2.1 as version

**Comment by whedon** [comment]
OK. v0.2.1 is the version.

**Comment by kbarnhart** [comment]
@whedon set 10.5281/zenodo.4606706 as archive


note 10.5281/zenodo.4606706 is the DOI for this version and 10.5281/zenodo.4606705 is the DOI for all versions.

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4606706" target="_blank">10.5281/zenodo.4606706</a> is the archive.

**Comment by kbarnhart** [comment]
@whedon recommend accept

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by kbarnhart** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.32614/RJ-2018-009 is OK
- 10.2830/63132 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2144

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2144, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kbarnhart** [comment]
@kadyb I've now recommended that this submission be accepted and published. One of the JOSS editors in chief @openjournals/joss-eics will handle the submission from here. Congratulations. 

Thanks to @adamhsparks and @mikerspencer for contributing thoughtful and thorough reviews.

**Comment by arfon** [comment]
@kadyb - I've made a few formatting fixes and small language changes in https://github.com/kadyb/rgugik/pull/74 - please let me know what you think.

Also, the footnotes on the first page were overflowing the margin of the paper so I moved them to links. Hopefully this is OK? If not, an alternative would be to make these proper references in the BibTeX.

<img width="746" alt="Screen Shot 2021-03-16 at 10 51 47" src="https://user-images.githubusercontent.com/4483/111297759-e6530d00-8645-11eb-93d4-dd40ee0d534d.png">

**Comment by kadyb** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02948/joss.02948/10.21105.joss.02948.pdf) :page_facing_up: :point_left:

**Comment by kadyb** [comment]
@arfon, all these changes are fine for us. Thanks!

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.32614/RJ-2018-009 is OK
- 10.2830/63132 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2146

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2146, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1371856343028629504) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2149
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02948](https://doi.org/10.21105/joss.02948)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@adamhsparks, @mikerspencer - many thanks for your reviews here and to @kbarnhart for editing this submission! JOSS relies upon the volunteer effort of folks like yourselves and we simply wouldn't be able to do this without you ✨ 

@kadyb - your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02948/status.svg)](https://doi.org/10.21105/joss.02948)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02948">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02948/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02948/status.svg
   :target: https://doi.org/10.21105/joss.02948
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02948/status.svg)](https://doi.org/10.21105/joss.02948)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by kadyb** [comment]
Thanks again to everyone for participating and helping in this submission!
